### DISCUSSION TITLE
Same padding, different results.

### DISCUSSION THREAD

**Discussion by rolandgvc** [google/flax#561] [Q&A] [answered]
Was there a change in how padding is being done? I remember passing 'SAME' as an argument to padding was equivalent to [(1,1),(1,1)] when using strides of 2. Can anyone clarify please?
```python
from flax import linen as nn
import numpy as np
x = np.random.rand(1,100,100, 3)
a = nn.max_pool(x, (3, 3), strides=(2, 2), padding='SAME')
b = nn.max_pool(x, (3, 3), strides=(2, 2), padding=[(1,1),(1,1)])
print(np.mean(np.abs(a - b))) # 0.068035774
```

**Comment by marcvanzee** [comment, marked as answer]
`SAME` pads so the input and output shape are the same. Given that 100 - 33*3 = 1, it only pads 1 after each spatial dimension. Right now your padding mask in `b` pads one before and one after, so I think it is doing too much padding.

```
>>> b2 = nn.max_pool(x, (3, 3), strides=(2, 2), padding=[(0, 1),(0, 1)])
>>> print(np.mean(np.abs(a - b2)))
0.0
```

**Reply by avital** [reply]
Curiously, this means that in your case, whether `padding='SAME'` is the same as `padding=[(1,1),(1,1)]` is dependent on whether the input width and height are even or odd

**Reply by rolandgvc** [reply]
I see now, thanks.
